CrossSectionAnalysis/CrossSectionAnalysis.py

```diff
@@ -0,0 +1,422 @@
+"""
+    # TODO: replace with organization, grant and thanks
+    self.parent.acknowledgementText = """
+This file was originally developed by Jean-Christophe Fillion-Robin, Kitware Inc., Andras Lasso, PerkLab,
+and Steve Pieper, Isomics, Inc. and was partially funded by NIH grant 3P41RR013218-12S1.
+"""
+
+#
+# CrossSectionAnalysisWidget
+#
+
+class CrossSectionAnalysisWidget(ScriptedLoadableModuleWidget, VTKObservationMixin):
+  """Uses ScriptedLoadableModuleWidget base class, available at:
+  https://github.com/Slicer/Slicer/blob/master/Base/Python/slicer/ScriptedLoadableModule.py
+  """
+
+  def __init__(self, parent=None):
+    """
+    Called when the user opens the module the first time and the widget is initialized.
+    """
+    ScriptedLoadableModuleWidget.__init__(self, parent)
+    VTKObservationMixin.__init__(self)  # needed for parameter node observation
+    self.logic = None
+    # Widget level observers to update module UI. Logic class has its own.
+    self.widgetMarkupPointObserver = None
+    self.widgetMarkupPointAddedObserver = None
+    self.widgetMarkupPointRemovedObserver = None
+    # Remove observers on previous path when currrent node has changed
+    self.previousPath = None
+
+  def setup(self):
+    """
+    Called when the user opens the module the first time and the widget is initialized.
+    """
+    ScriptedLoadableModuleWidget.setup(self)
+
+    # Load widget from .ui file (created by Qt Designer).
+    # Additional widgets can be instantiated manually and added to self.layout.
+    uiWidget = slicer.util.loadUI(self.resourcePath('UI/CrossSectionAnalysis.ui'))
+    self.layout.addWidget(uiWidget)
+    self.ui = slicer.util.childWidgetVariables(uiWidget)
+
+    # Set scene in MRML widgets. Make sure that in Qt designer the top-level qMRMLWidget's
+    # "mrmlSceneChanged(vtkMRMLScene*)" signal in is connected to each MRML widget's.
+    # "setMRMLScene(vtkMRMLScene*)" slot.
+    uiWidget.setMRMLScene(slicer.mrmlScene)
+
+    # Create logic class. Logic implements all computations that should be possible to run
+    # in batch mode, without a graphical user interface.
+    self.logic = CrossSectionAnalysisLogic()
+    
+    # Hide diameter labels. Concern only VMTK centerline models.
+    self.showDiameterLabels(False)
+    
+    self.ui.moreCollapsibleButton.collapsed = True
+    self.ui.advancedCollapsibleButton.collapsed = True
+    self.ui.roiCollapsibleButton.collapsed = True
+    slicer.modules.reformat.widgetRepresentation().setEditedNode(slicer.util.getNode("vtkMRMLSliceNodeRed"))
+    self.resetSliderWidget()
+
+    # Connections
+    self.ui.inputSelector.connect("currentNodeChanged(vtkMRMLNode*)", self.onSelectNode)
+    self.ui.positionIndexSliderWidget.connect("valueChanged(double)", self.logic.process)
+    # Feedback on module UI
+    self.ui.positionIndexSliderWidget.connect("valueChanged(double)", self.showCurrentPositionData)
```

Comment:
This one is difficult to understand. I grant that you are speaking of multiple loaded volume nodes.

The module works on the 2D view selected by a radio button. If the curve is created according to one data volume, and the user changes the background volume of that 2D view, he knows what he is doing.

Or perhaps I have not fully understood the issue.

Reply:
There are many layouts where you have not just red, green, yellow slice views.

If you use a qMRMLNodeComboBox instead of radio buttons then you can easily show the list of all slice view nodes.

Reply:
I really don't see how it is benefiting. Putting a qMRMLNodeComboBox listing Red, Green and Yellow is easy. But whatever the layout, whatever the way we specify a 2D slice node, we'll still be selecting between Red, Green and Yellow. Visual result will always be in one of these, whether it is visible or not. I'll leave this one for now.

Reply:
The combobox will automatically display all the slice views, not just red, green, and yellow.

Reply:
OK, got it. There can be more 2D views than Red, Green and Yellow. This is well demonstrated in Landmark Registration.

Using now a qMRMLNodeComboBox to list all 2D slice nodes, instead of named RGY views.